src/compiler/scala/tools/nsc/transform/Erasure.scala

```diff
@@ -805,6 +805,11 @@ abstract class Erasure extends InfoTransform
                 return result setType ErasedValueType(tref.sym, result.tpe)
 
             }
+          case tree @ Block(_, _: LabelDef) =>
+            // Avoid pushing the adaptation to a matchEnd label.
+            // This helps the back-end to generate better code.
+            val block1 = super.typed1(adaptMember(tree), mode, WildcardType)
+            adapt(block1, mode, pt, tree)
```

Comment:
why is the call to `adapt` needed? doesn't the outer `typed` call invoke `adapt` anyway?

Reply:
Good question. I'll try removing it and see if tests still pass.

Reply:
Everything passes locally, so I pushed the alternative that doesn't call `adapt` here. And also used your suggestion as comment.